client/client.go

```diff
@@ -107,7 +107,7 @@ func CheckRedirect(req *http.Request, via []*http.Request) error {
 // Use DOCKER_API_VERSION to set the version of the API to reach, leave empty for latest.
 // Use DOCKER_CERT_PATH to load the TLS certificates from.
 // Use DOCKER_TLS_VERIFY to enable or disable TLS verification, off by default.
-func NewEnvClient() (*Client, error) {
+func NewEnvClient(ops ...func(*Client)) (*Client, error) {
```

Comment:
These changes on `client` are not strictly required (and might be better off discussed in a separate PR)

Reply:
I think we've wanted to make this change for a while. What do you think about something like this:

* create a `func FromEnv(*Client)` and move all the code from `NewEnvClient` (except the `ops` addition) to `FromEnv()`
* create `NewClientWithOpts()` (or something like that) which creates a client from ops, but has no other inline setup.
* remove `ops`  from `NewEnvClient` and deprecate it. Replace the implementation with `NewClientWithOpts(FromEnv)`.

That way the new interface would create a bare `client{}` and we can apply any ops onto it. If there are required fields (host and client?) maybe it would also validate that.

Reply:
sounds very good to me 👼